x-pack/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_actions.ts

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { DeleteByQueryResponse } from '@elastic/elasticsearch/api/types';
+import { HostMetadata } from '../types';
+import {
+  EndpointActionGenerator,
+  LogsEndpointAction,
+  LogsEndpointActionResponse,
+} from '../data_generators/endpoint_action_generator';
+import { wrapErrorAndRejectPromise } from './utils';
+
+const ENDPOINT_ACTIONS_INDEX = '.logs-endpoint.actions-default';
+const ENDPOINT_ACTION_RESPONSES_INDEX = '.logs-endpoint.action.responses-default';
+
+const defaultEndpointActionGenerator = new EndpointActionGenerator();
+
+export interface IndexedEndpointActionsForHostResponse {
+  endpointActions: LogsEndpointAction[];
+  endpointActionResponses: LogsEndpointActionResponse[];
+  endpointActionsIndex: string;
+  endpointActionResponsesIndex: string;
+}
+
+/**
+ * Indexes a random number of Endpoint Actions for a given host
+ *
+ * @param esClient
+ * @param endpointHost
+ * @param [endpointActionGenerator]
+ */
+export const indexEndpointActionsForHost = async (
+  esClient: Client,
+  endpointHost: HostMetadata,
+  endpointActionGenerator: EndpointActionGenerator = defaultEndpointActionGenerator
+): Promise<IndexedEndpointActionsForHostResponse> => {
+  const agentId = endpointHost.elastic.agent.id;
+  const total = endpointActionGenerator.randomN(5);
+  const response: IndexedEndpointActionsForHostResponse = {
+    endpointActions: [],
+    endpointActionResponses: [],
+    endpointActionsIndex: ENDPOINT_ACTIONS_INDEX,
+    endpointActionResponsesIndex: ENDPOINT_ACTION_RESPONSES_INDEX,
+  };
+
+  for (let i = 0; i < total; i++) {
+    // create an action
+    const action = endpointActionGenerator.generate({
+      EndpointAction: {
+        data: { comment: 'data generator: this host is same as bad' },
+      },
+    });
+
+    action.agent.id = [agentId];
+
+    esClient
```

Comment:
Do you mean the error message won't be seen?

Reply:
correct. Not unless you `await` or add a `.catch()` that some how makes the error visisble - which I'm still not seeing how that can be done. 

I think you really just forgot to `await` these api calls

Reply:
But there is a `catch` block that is chained here. Same with the other one.

Reply:
yes, but take alook at that .catch() handler. All it does is wrap the error and then returns a `Promise.reject`. With promises the error need to be handled explicitly, unless you are in a `async` function and add in you own `try {} catch` and use `await`. 

I have not checked this out, but try this to see what happens - add this after the `index()` method

```typescript
.then(() => { throw new Error('crap'); } )
```

so that it looks like this:

```typescript
esClient
  .index({
      index: ENDPOINT_ACTIONS_INDEX,
      body: action,
    })
    .then(() => { throw new Error('crap'); } )
    .catch(wrapErrorAndRejectPromise);
```

This will result in a rejected Promise for the `index()` call, but it will not reject your `indexEndpointActionsForHost()` function call - it will keep going because you did not `await` and handle the error. 


Here is a quick example from the browser console:

![image](https://user-images.githubusercontent.com/56442535/135308015-b5659a7f-3766-442a-ae9c-4b37750090f7.png)

Reply:
Ah, I see! Thanks for explaining this. 🙏

Reply:
done a23277db2db04227d6e2fd7a48145f487d0e87fd 😄